src/components/ThemeToggle.tsx

```diff
@@ -0,0 +1,37 @@
+import React, {useCallback, useEffect, useState} from 'react';
+import halfMoon from "../assets/half-moon.png"
+import sheep from "../assets/sheep.png"
+
+const ThemeToggle = () => {
+  const [isDark, setIsDark] = useState(false)
```

Comment:
I don't like using a boolean for this. A better approach might be using an enum or a union type (`type Theme = "light" | "dark"`).